### PR TITLE
For non-authorised users OrganizationView not working

### LINKED ISSUE #2148: For non-authorised users OrganizationView is absolutely not working
1) Tools are not loading
2) Tests are not loading
3) Users are not shown, avatar is not shown

https://github.com/saveourtool/save-cloud/assets/58667063/f98b6de4-5851-4038-bd29-18cee9e593cb

### PR DESCRIPTION
**Pull request by Cheshiriks**
For non-authorised users OrganizationView is absolutely not working

Closes #2148

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt [hunk 1]
```diff
@@ -109,16 +107,18 @@ class LnkUserOrganizationController(
...
+        getUserAndOrganizationWithPermissions(
+            userName ?: authentication.toUser().name!!,
+            organizationName,
+            Permission.READ,
+            authentication,
+        )
+            .map { (user, organization) ->
+                lnkUserOrganizationService.getRole(user, organization)
+            }
+    } ?: Role.NONE.toMono()
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt:121]: Do we have role viewer?

Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt:121]: Viewer role for authorized users who are not members of this organization.

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt [hunk 2]
```diff
@@ -109,16 +107,18 @@ class LnkUserOrganizationController(
...
-        Permission.READ,
-        authentication,
-    )
-        .map { (user, organization) ->
-            lnkUserOrganizationService.getRole(user, organization)
-        }
+        authentication: Authentication?,
+    ): Mono<Role> = authentication?.let {
+        getUserAndOrganizationWithPermissions(
+            userName ?: authentication.toUser().name!!,
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt:113]: `userName` can be provided but `authentication` can be `null` and it still leads to `Role.NONE`
```suggestion
        authentication: Authentication?,
    ): Mono<Role> = (userName ?: authentication?.toUser()?.name)?.let { _userName ->
        getUserAndOrganizationWithPermissions(
            _userName,
            organizationName,
            Permission.READ,
            authentication,
        )
            .map { (user, organization) ->
                lnkUserOrganizationService.getRole(user, organization)
            }
    } ?: Role.NONE.toMono()
```

Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt:113]: Can we have a UserName for non-authenticated users? After all, we take this name just from the database after authentication

Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt:113]: userName param is _required = false_. Somewhere in the code we use userName to call the controller, but somewhere we don’t pass anything and use authentication. I don't know why it was implemented like this. But if you remove _authentication_, then the logic will break.

Comment (Cheshiriks) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkUserOrganizationController.kt]: Removed UserName

**Review by nulls — APPROVED** [review]

**Pull request merged**